Source/WebCore/page/Navigation.cpp

```diff
@@ -613,6 +614,50 @@ void Navigation::abortOngoingNavigation(NavigateEvent& event)
     }
 }
 
+static std::pair<Ref<DOMPromise>, Ref<DeferredPromise>> createPromiseAndWrapper(Document& document)
```

Comment:
At some point maybe we want to share this somewhere, I think I got it from ViewTransitions but IIRC there is even one more user.

Reply:
Yeah, please refactor rather than copy/paste code.